### PR TITLE
Bug 2089224: Remove monitoring config reconciliation

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Instead of hypershift syncing in a config into the guest cluster, the
CMO should default to the proper node selector for the prometheus
operator deployment. This allows the config inside the guest cluster to
remain entirely user driven.

This commit removes the code to reconcile the configuration.

**Which issue(s) this PR fixes** 
Fixes #2089224

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Comment by csrwng** [comment]
/hold
We first need a change in the CMO that will set the default node selector as in https://github.com/openshift/cluster-monitoring-operator/pull/1679

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go [hunk 1]
```diff
@@ -1,45 +0,0 @@
-package monitoring
-
-import (
-	"fmt"
-
-	cmoconfig "github.com/openshift/hypershift/thirdparty/clustermonitoringoperator/pkg/manifests"
```
Comment (alvaroaleman) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go:6]: Can we delete this package now that it is unused?

Comment (csrwng) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go:6]: we still have another file in there
https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/servicemonitor.go

Comment (alvaroaleman) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go:6]: I meant the github.com/openshift/hypershift/thirdparty/clustermonitoringoperator/pkg/manifests package

Comment (csrwng) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go:6]: (which btw has another bug :) )
I just realized we're only keeping etcd_object_counts ... it really should look like the standalone kube-apiserver service monitor

Comment (csrwng) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/monitoring/config.go:6]: ah, yes, thx

**Comment by csrwng** [comment]
Looks like there's more stuff to fix up:
```
E0527 02:31:24.458727       1 run.go:74] "command failed" err="invalid argument \"CSIMigrationOpenStack=false\" 
for \"--feature-gates\" flag: cannot set feature gate CSIMigrationOpenStack to false, feature is locked to true"
```

**Comment by csrwng** [comment]
/hold cancel
The CMO change is now in

**Pull request merged**